sympy/core/evalf.py

```diff
@@ -365,22 +365,11 @@ def calc_part(re_im, nexpr):
             # expression
             s = options.get('subs', False)
             if s:
-                doit = True
-                from sympy.core.compatibility import as_int
-                # use strict=False with as_int because we take
-                # 2.0 == 2
-                for v in s.values():
-                    try:
-                        as_int(v, strict=False)
-                    except ValueError:
-                        try:
-                            [as_int(i, strict=False) for i in v.as_real_imag()]
-                            continue
-                        except (ValueError, AttributeError):
-                            doit = False
-                            break
-                if doit:
-                    re_im = re_im.subs(s)
+                re_im = re_im.subs(s)
+                # XXX: There was a load more code here that was added in
+                # https://github.com/sympy/sympy/pull/10346
+                # I guess something has changed because now I can delete it
+                # without apparently affecting anything...
```

Comment:
CC @smichr

Reply:
Interesting because I just added the comment about 2.0 == 2 in a recent PR. I recall that the test was written with 2.0 on the rhs even though it should be 2 from this code. (It would be *nice* if it were 2.0 but I don't think it was.) The try/excepts are pretty limited here: the attribute error is for `as_real_imag` and the ValueError is for the value not being an int. I'll see if I can find the test...

Reply:
ok, #16081 and it was line 81 of `def test_powers` that had `assert (x**(4.0*y)).subs(x**(2.0*y), n) == n**2.0`

I would be very tentative in removing this without making sure that tests traverse this code for the condition for which it was written.

Reply:
This has now passed the entire test suite on Travis.

I've locally checked out #10346 (where this was added) and I can apply the same change there and run the tests that were added there and they pass:
```console
$ git checkout pr/10346
Branch 'pr/10346' set up to track remote branch 'pr/10346' from 'upstream'.
Switched to a new branch 'pr/10346'
$ git stash apply
On branch pr/10346
Your branch is up to date with 'upstream/pr/10346'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   sympy/core/evalf.py

no changes added to commit (use "git add" and/or "git commit -a")
$ git diff
diff --git a/sympy/core/evalf.py b/sympy/core/evalf.py
index c568ae1277..4f9ca43404 100644
--- a/sympy/core/evalf.py
+++ b/sympy/core/evalf.py
@@ -328,20 +328,7 @@ def calc_part(expr, nexpr):
             # expression
             s = options.get('subs', False)
             if s:
-                doit = True
-                from sympy.core.compatibility import as_int
-                for v in s.values():
-                    try:
-                        as_int(v)
-                    except ValueError:
-                        try:
-                            [as_int(i) for i in v.as_real_imag()]
-                            continue
-                        except (ValueError, AttributeError):
-                            doit = False
-                            break
-                if doit:
-                    expr = expr.subs(s)
+                expr = expr.subs(s)
 
             expr = Add(expr, -nint, evaluate=False)
             x, _, x_acc, _ = evalf(expr, 10, options)
$ bin/test sympy/core/tests/test_evalf.py -k test_evalf_integer_parts test_issue_10323 -v
==================================================================== test process starts =====================================================================
executable:         /Users/enojb/current/sympy/27venv/bin/python  (2.7.10-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
random seed:        61558505
hash randomization: on (PYTHONHASHSEED=1592196562)

sympy/core/tests/test_evalf.py[2] 
test_evalf_integer_parts ok
test_issue_10323 ok                                                                                                                                       [OK]


========================================================= tests finished: 2 passed, in 0.93 seconds ==========================================================
```
I think maybe the other changes in that PR fixed the problem and this must have seemed necessary at some point while working on it but wasn't by the end.

Reply:
Should I restore this code and put this change in a separate PR?

Reply:
I guess not. It should have been tested and if it was and passes without this code, something else is working around it.

Reply:
On second thought...the `evalf` machinery should be handled very carefully. The subs cannot be done outright unless only Integers are involved (or, as this part is testing, integer real and imaginary parts). When integer (or rational? but maybe the expression at this point is split into numer and denom) the calculation can proceed in an exact manner, otherwise the precision of "subends" needs to be increased as the overall precision of the computed object is increased. If they get subs so this can't happen, you can get hard to track down evaluation failures.

Reply:
I'll split this example into a separate PR but I think that we should try to find an example that triggers this code path or otherwise remove it.

One cause of hard to track down failures is overly broad exception handling:
[https://github.com/sympy/sympy/pull/15935#issuecomment-461132761](https://github.com/sympy/sympy/pull/15935#issuecomment-461132761)

Reply:
I've reverted this change from this PR and opened it as #16175